workflow/executor/resource.go

```diff
@@ -40,22 +41,25 @@ func (we *WorkflowExecutor) ExecResource(action string, manifestPath string, fla
 	if err != nil {
 		exErr := err.(*exec.ExitError)
 		errMsg := strings.TrimSpace(string(exErr.Stderr))
-		return "", "", errors.New(errors.CodeBadRequest, errMsg)
+		return "", "", "", schema.GroupVersionResource{}, errors.New(errors.CodeBadRequest, errMsg)
 	}
 	if action == "delete" {
-		return "", "", nil
+		return "", "", "", schema.GroupVersionResource{}, nil
 	}
 	if action == "get" && len(out) == 0 {
-		return "", "", nil
+		return "", "", "", schema.GroupVersionResource{}, nil
 	}
 	obj := unstructured.Unstructured{}
 	err = json.Unmarshal(out, &obj)
 	if err != nil {
-		return "", "", err
+		return "", "", "", schema.GroupVersionResource{}, err
 	}
-	resourceName := fmt.Sprintf("%s.%s/%s", obj.GroupVersionKind().Kind, obj.GroupVersionKind().Group, obj.GetName())
-	log.Infof("%s/%s", obj.GetNamespace(), resourceName)
-	return obj.GetNamespace(), resourceName, nil
+	resourceFullName := fmt.Sprintf("%s.%s/%s", obj.GroupVersionKind().Kind, obj.GroupVersionKind().Group, obj.GetName())
+	log.Infof("%s/%s", obj.GetNamespace(), resourceFullName)
+	selfLink := strings.Split(obj.GetSelfLink(), "/")
```

Comment:
ek! we have to guess the resource name - was that a problem with the previous method?

Reply:
Nope. Previous method only needed a selfLink! But this is one line should be stable as selfLink is auto populated by k8s with its convention.